### ISSUE TITLE
MultiModal does not work with Next.js Frontend and FastAPI backend

### ISSUE THREAD

**Issue by Laktus** [run-llama/create-llama#65]
Hi,

I wanted to implement custom evaluating logic. Realizing that only the python implemention of LLamaIndex supports `QuestionGenerator` i thought that it would be more reasonable to the FastAPI backend + Next.js Frontend setup.

I managed to pass the data for images to the backend extending the handleSubmit of useChat for https://github.com/vercel/ai/pull/725. I however don't know how to duplicate the functionality of StreamData in the FastAPI backend.

Can you make this example work out of the box or provide some further documentation of how to implement this? Currently the multi modality does not work, without multiple changes.

Thanks for taking your time and reading my request.

**Comment by marcusschiesser** [comment]
@Laktus by coincidence, I just added a vercel/ai compatible StreamingResponse in the last release, see https://github.com/run-llama/create-llama/blob/main/templates/types/streaming/fastapi/app/api/routers/vercel_response.py

**Comment by Laktus** [comment]
@marcusschiesser That looks awesome thanks! I think someone still needs to modify the template to work with it though. Or is it already integrated in the last release? The template is missing being able to handle the incoming data from `useChat`'s `handleSubmit` as well as passing it back from server to frontend using the `StreamingTextResponse` (or your `vercel_response` in FastAPI).

**Comment by marcusschiesser** [comment]
@Laktus The template is part of create-llama since npx create-llama@0.1.0 - It was just updated in npx create-llama@0.1.1 - what are you missing?

**Comment by Laktus** [comment]
@marcusschiesser I will try integrating my changes to the backend for handling the data parameter and then will add a message if it works, thanks for the update!

**Comment by Laktus** [comment]
@marcusschiesser 
Hi Marcus, i don't see any inherent integration with images in the FastAPI backend. Do you know how i can add this? How do i pass image information into the ChatMessage object when im using a image capable model like GPT-4 or GPT-4-Vision? (I already managed to pass the image from front-to-backend and back to the frontend for the display)

Thanks for any help.

**Comment by marcusschiesser** [comment]
@Laktus, the problem is that in Python, you have to use the `MultiModalVectorStoreIndex` to use images

So I would start replacing `VectorStoreIndex` with this class.

Details about using it are here:
https://docs.llamaindex.ai/en/stable/examples/evaluation/multi_modal/multi_modal_rag_evaluation/?h=multimodalvectorstoreindex

If you like, you're welcome to post a diff of your code here.

**Comment by marcusschiesser** [comment]
@Laktus yes, this is a current issue of the Python version. We're working on aligning the multi-modal capabilities of the Python and the Typescript version. Once that's done, we will add image upload support to the FastAPI backend

**Comment by Laktus** [comment]
@marcusschiesser Is there any update on when this will be implemented?

**Comment by marcusschiesser** [comment]
Not yet, we'll need multi-modal support in the Python framework first
